build/templates/session.py.mako

```diff
@@ -143,6 +145,13 @@ class Session(object):
         self.library = library.get_library()
         self.vi = 0  # This must be set before calling _init_with_options.
         self.vi = self._init_with_options(resource_name, id_query, reset_device, options_string)
+
+        self.__isfrozen = True
+
+    def __setattr__(self, key, value):
+        if self.__isfrozen and key not in dir(self):
+            raise TypeError("%r is a frozen class" % self)
```

Comment:
https://docs.python.org/2/library/exceptions.html#exceptions.AttributeError
exception AttributeError
Raised when an attribute reference (see Attribute references) or assignment fails. (When an object does not support attribute references or attribute assignments at all, TypeError is raised.)

https://docs.python.org/2/library/exceptions.html#exceptions.TypeError
exception TypeError
Raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch.

https://docs.python.org/2/library/exceptions.html#exceptions.KeyError
exception KeyError
Raised when a mapping (dictionary) key is not found in the set of existing keys.

I suppose TypeError is the right one after all! My bad.